### DISCUSSION TITLE
Access defaultValues from useFormState()

### DISCUSSION THREAD

**Discussion by ghost** [react-hook-form/react-hook-form#10799] [General] [answered]
Hello,

I find this https://github.com/react-hook-form/react-hook-form/issues/1044 issue, and seems the defaultValues is not exposed to user.

But by reading the documentation, I found it shows defaultValues is available from useFormState(). The implementation and documentation seems not matching.

https://react-hook-form.com/docs/useformstate
<img width="1042" alt="image" src="https://github.com/react-hook-form/react-hook-form/assets/81522010/f4a0ba3a-4f1e-4125-a24b-0f13e6b5f705">

By the way, it would be very helpful if users can access the defaultValues from react-hook-form API.
For example, somewhere down the component tree, user can access the defaultValues and use it to reset the whole form state to the initial.

Many thanks!

**Comment by leapful** [comment, marked as answer]
@yaobao1993 

If you read carefully the discussion https://github.com/react-hook-form/react-hook-form/issues/1044 that you mentioned previously, you will see that the discussion was created in year 2020 and it's currently talking about `useFormContext()` hook. At that time, for sure, `useFormState()` hook did not exist and the implementation of react-hook-form in 2020 is definitely not matching with documentation in 2023.

Please try `defaultValues` with `useFormState()` hook first and if it does not work, please provide a minimal re-producible codesandbox with steps to re-produce on this discussion.

Example: https://codesandbox.io/s/useformstate-forked-q86ml8?file=/src/App.js

**Comment by ghost** [comment]
After some investigation, I found that we are using the older version of react-hook-form before this feature get introduced (Version 7.37.0).

We will use the newest version of react-hook-form.

Thanks for the helpful reply!
